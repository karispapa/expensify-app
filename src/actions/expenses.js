import uuid from 'uuid';


//ADD_EXPENSE
// action generator for add expense
export const addExpense = ({
  description = '',
  text = '',
  amount =0,
createdAt = 0}
  ={})=> ({
    type: 'ADD_EXPENSE',
    expense: {
      id: uuid(),
      description,
      text,
      amount,
      createdAt
    }
});

//REMOVE_EXPENSE
// action generator for remove expense
export const removeExpense = ({id} = {})=> ({
  type: 'REMOVE_EXPENSE',
  id
});
//EDIT_EXPENSE
// action generator for edit expense
export const editExpense = (id, updates)=> ({
  type: 'EDIT_EXPENSE',
  id,
  updates
});