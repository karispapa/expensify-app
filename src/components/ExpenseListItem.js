import React from 'react';
import {Link} from 'react-router-dom'
// import { connect } from 'react-redux';
// import { removeExpense } from '../actions/expenses'

const ExportListItem = ({id, description, amount, createdAt})=>(

  <div>
    <Link to={`/edit/${id}`}>
     <h3>{description} </h3>
    </Link>
    
    <p> {createdAt} ksh: {amount} </p>
  </div>
);

export default ExportListItem; 