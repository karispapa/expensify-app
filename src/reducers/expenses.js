

const expensesReducerDefault = [];

// Expenses reducer

export default (state = expensesReducerDefault, action)=>{

  switch(action.type){

    case 'ADD_EXPENSE':
      return [...state, action.expense]

    case 'REMOVE_EXPENSE':
      return state.filter(({id})=> action.id !== id) // filter returns all items in the array that pass the set criteria

     case 'EDIT_EXPENSE': 
     return state.map((expense)=>{
       if(expense.id === action.id){
          return {
            ...expense,
            ...action.updates
          }
       } else{
          return expense
       }
     });
    default: 
      return state
  }
};