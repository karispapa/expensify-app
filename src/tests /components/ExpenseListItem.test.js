import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'

test ('Should render an expense from the expense array', ()=>{
  const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>)
  expect(wrapper).toMatchSnapshot();
})