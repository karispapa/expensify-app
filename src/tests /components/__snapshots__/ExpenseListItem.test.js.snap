// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an expense from the expense array 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      lunch
       
    </h3>
  </Link>
  <p>
    $10,000.00
    -
    January 1st, 1970
  </p>
</div>
`;
